src/discord/Bot.ts

```
@@ -38,28 +38,28 @@ export default class Bot {
         });
     });
 
-    this.client.on('message', this.messageHandle);
+    this.client.on('message', (message) => {
+      this.messageHandle(message).then(() => { }).catch(console.error);
+    });
 
     this.client.on('messageReactionAdd', this.messageReactionAddHandle);
     this.client.on('messageReactionRemove', this.messageReactionRemoveHandle);
   }
 
-  messageHandle = (message: Message) => {
+  messageHandle = async (message: Message) => {
     if (message.author.bot) return;
     if (!message.content.startsWith(this.config.prefix)) {
       if (message.channel.id === channels.introductions // Message is in the introductions channel
         && message.content.includes(' ') // Message is longer than one word
         && message.member // Person is real
         && !(message.member.roles.cache.has('742797850630684762'))) { // ID of "member" role
-        message.member.roles.add('742797850630684762').then(() => {
+        try {
```

You actually don't want to try catch locally. Throw an error and let it bubble all the way up to the message handler's catch.